### PR TITLE
Feat:  Add --mode option to sqlemsh ui CLI command to define modules

### PR DESCRIPTION
**Pull request by mykalmax**
Introducing `--mode` flag to `sqlmesh ui` command. Different modes load different UI modules.
List of modules: 
- EDITOR
- FILES
- DOCS
- ERRORS
- PLANS

List of modes:
- IDE (loads all available modules)
- DOCS (loads only docs and errors)
- PLAN (loads only plan and errors)
- DEFAULT (PLAN + DOCS, currently set as default when running )

Currently `DEFAULT` is default mode when running `sqlmesh ui` and `IDE` when `make ui-up`

### CODE REVIEW

File: sqlmesh/cli/main.py [hunk 1]
```diff
@@ -562,13 +562,15 @@ def info(obj: Context) -> None:
     default=8000,
     help="Bind socket to this port. Default: 8000",
 )
+@click.option(
+    "--mode",
+    type=str,
+    default="all",
```
Comment (vchan) [inline sqlmesh/cli/main.py:568]: Is there an "all" mode? Has that become "ide"?

File: web/server/settings.py [hunk 2]
```diff
@@ -25,6 +39,12 @@ class Settings(PydanticModel):
     )
     config: str = Field(default_factory=lambda: os.getenv("CONFIG", ""))
     gateway: t.Optional[str] = Field(default_factory=lambda: os.getenv("GATEWAY"))
+    ui_mode: str = Field(default_factory=lambda: os.getenv("UI_MODE", Mode.DEFAULT.value))
```
Comment (vchan) [inline web/server/settings.py:42]:
```suggestion
    ui_mode: Mode = Field(default_factory=lambda: os.getenv("UI_MODE", Mode.DEFAULT.value))
```

File: web/server/settings.py [hunk 3]
```diff
@@ -25,6 +39,12 @@ class Settings(PydanticModel):
...
+
+    @property
+    def modules(self) -> t.Set[models.Modules]:
+        mode = self.ui_mode if self.ui_mode in mode_to_modules else models.Mode.IDE.value
```
Comment (vchan) [inline web/server/settings.py:46]: If the type annotation for ui_mode is Mode, then do you still need this check?

File: web/client/src/library/components/loading/LoadingSegment.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  return (
+    <div
+      className={clsx(
+        'flex justify-center items-center w-full h-full',
+        className,
+      )}
+    >
+      <Loading className="inline-block">
+        <Spinner className="w-3 h-3 border border-neutral-10 mr-4" />
+        <h3 className="text-md">{children}.</h3>
```
Comment (vchan) [inline web/client/src/library/components/loading/LoadingSegment.tsx:22]: Do you always want append the period to children?

File: web/client/src/library/pages/models/Models.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    const file = files.get(model.path)
+
+    if (isNotNil(file)) {
+      setSelectedFile(file)
+    }
+
+    setLastSelectedModel(model)
+
+    navigate(`${to}/${model.name}`, { replace: true })
+  }, [models, model, to])
```
Comment (vchan) [inline web/client/src/library/pages/models/Models.tsx:54]: Does `models` need to be a dependency?

File: web/client/src/models/module-controller.ts [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+  get isEmpty(): boolean {
+    return this.modules.size === 0
+  }
+
+  get hasErrors(): boolean {
+    return this.modules.has(Modules.errors)
+  }
+
+  get hasEidtor(): boolean {
```
Comment (vchan) [inline web/client/src/models/module-controller.ts:27]: hasEditor?

File: sqlmesh/cli/main.py [hunk 7]
```diff
@@ -562,13 +562,15 @@ def info(obj: Context) -> None:
     default=8000,
     help="Bind socket to this port. Default: 8000",
 )
+@click.option(
+    "--mode",
+    type=click.Choice(["ide", "default", "docs", "plan"], case_sensitive=False),
```
[the remark this answers is not in the document]
Comment (mykalmax) [reply sqlmesh/cli/main.py:567]: I just did not want to bring `Mode enum` form `server package` to `core`

File: web/client/src/library/components/environmentDetails/EnvironmentDetails.tsx [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+      isFalse(planAction.isProcessing) &&
+      isFalse(planAction.isRunningTask)
+    ) {
+      planOverview.reset()
+      planApply.reset()
+      planCancel.reset()
+
+      void planRun()
+    }
+  }, [models, environment])
```
[the remark this answers is not in the document]
Comment (mykalmax) [reply web/client/src/library/components/environmentDetails/EnvironmentDetails.tsx:62]: we want call planRun (or refresh state to see latest changes/backfills) when either models  changed or environment  (selected in this case) has changed

File: web/client/src/library/components/plan/PlanApplyStageTracker.tsx [hunk 9]
```diff
@@ -243,6 +243,7 @@ function StageEvaluate({
   const elStageEvaluate = useRef<HTMLDivElement>(null)
 
   useEffect(() => {
+    console.log('scrolling to evaluate')
```
Comment (vchan) [inline web/client/src/library/components/plan/PlanApplyStageTracker.tsx:246]:
```suggestion
```

File: web/client/src/library/components/plan/PlanActions.tsx [hunk 10]
```diff
@@ -105,6 +109,8 @@ export default function PlanActions({
     isFalse(planAction.isCancelling) &&
     isFalse(planAction.isDone) &&
     (planAction.isProcessing ? isFalse(planCancel.isSuccessed) : true)
+  const isFailedOrCanceled =
+    planOverview.isFailed || planApply.isFailed || planCancel.isSuccessed
```
Comment (vchan) [inline web/client/src/library/components/plan/PlanActions.tsx:113]: Nit: I just noticed this. Sorry I didn't catch it before. isSuccessful? hasSucceeded?

**Review by vchan — APPROVED** [review]

**Review by vchan — APPROVED** [review]

**Pull request merged**